wrappers/services_test.go

```diff
@@ -3180,6 +3221,60 @@ apps:
 	})
 }
 
+func (s *servicesTestSuite) TestRestartInDifferentStates(c *C) {
+	const manyServicesYaml = `name: test-snap
+version: 1.0
+apps:
+  svc1:
+    command: bin/foo
+    daemon: simple
+  svc2:
+    command: bin/foo
+    daemon: simple
+  svc3:
+    command: bin/foo
+    daemon: simple
+  svc4:
+    command: bin/foo
+    daemon: simple
+`
+	srvFile1 := "snap.test-snap.svc1.service"
+	srvFile2 := "snap.test-snap.svc2.service"
+	srvFile3 := "snap.test-snap.svc3.service"
+	srvFile4 := "snap.test-snap.svc4.service"
+
+	info := snaptest.MockSnap(c, manyServicesYaml, &snap.SideInfo{Revision: snap.R(1)})
+
+	r := systemd.MockSystemctl(func(cmd ...string) ([]byte, error) {
+		states := map[string]serviceState{
+			srvFile1: {"active", "enabled"},
+			srvFile2: {"inactive", "enabled"},
+			srvFile3: {"active", "disabled"},
+			srvFile4: {"inactive", "disabled"},
+		}
+		if out := mockSystemCtlShow(cmd, states); out != nil {
+			return out, nil
+		}
+		s.sysdLog = append(s.sysdLog, cmd)
```

Comment:
For clarity I think this should be arranged as follows:

```suggestion
		s.sysdLog = append(s.sysdLog, cmd)
		if out := mockSystemCtlShow(cmd, states); out != nil {
			return out, nil
		}
```

and then the log of commands will list `systemctl show` calls from Status() too.